specs/patch/fixtures/fixtures-1.md

```diff
@@ -0,0 +1,283 @@
+
+
+### Inserting into a list
+
+Given an initial document:
+
+[testmark]:# (inserting-into-a-list/initial)
+```json
+["bar", "baz"]
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (inserting-into-a-list/patch)
+```json
+[
+	{ "op": "add", "path": "/1", "value": "qux" }
+]
+```
+
+The following document should result:
+
+[testmark]:# (inserting-into-a-list/result)
+```json
+[
+	"bar",
+	"qux",
+	"baz"
+]
+```
+
+
+### Removing an entry from a map
+
+Given an initial document:
+
+[testmark]:# (removing-map-entry/initial)
+```json
+{
+	"baz": "qux",
+	"foo": "bar"
+}
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (removing-map-entry/patch)
+```json
+[
+	{ "op": "remove", "path": "/baz" }
+]
+```
+
+The following document should result:
+
+[testmark]:# (removing-map-entry/result)
+```json
+{
+	"foo": "bar"
+}
+```
+
+
+### Replacing an entry from a map
+
+Given an initial document:
+
+[testmark]:# (replacing-map-entry/initial)
+```json
+{
+	"baz": "qux",
+	"foo": "bar"
+}
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (replacing-map-entry/patch)
+```json
+[
+	{ "op": "replace", "path": "/baz", "value": "boo" }
+]
+```
+
+The following document should result:
+
+[testmark]:# (replacing-map-entry/result)
+```json
+{
+	"baz": "boo",
+	"foo": "bar"
+}
+```
+
+
+### Copying a value
+
+Given an initial document:
+
+[testmark]:# (copy/initial)
+```json
+{
+	"foo": {
+		"bar": "baz",
+		"waldo": "fred"
+	},
+	"qux": {
+		"corge": "grault"
+	}
+}
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (copy/patch)
+```json
+[
+	{ "op": "copy", "from": "/foo/waldo", "path": "/qux/thud" }
+]
+```
+
+The following document should result:
+
+[testmark]:# (copy/result)
+```json
+{
+	"foo": {
+		"bar": "baz",
+		"waldo": "fred"
+	},
+	"qux": {
+		"corge": "grault",
+		"thud": "fred"
+	}
+}
+```
+
+
+### Moving a value
+
+Given an initial document:
+
+[testmark]:# (move/initial)
+```json
+{
+	"foo": {
+		"bar": "baz",
+		"waldo": "fred"
+	},
+	"qux": {
+		"corge": "grault"
+	}
+}
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (move/patch)
+```json
+[
+	{ "op": "move", "from": "/foo/waldo", "path": "/qux/thud" }
+]
+```
+
+The following document should result:
+
+[testmark]:# (move/result)
+```json
+{
+	"foo": {
+		"bar": "baz"
+	},
+	"qux": {
+		"corge": "grault",
+		"thud": "fred"
+	}
+}
+```
+
+
+### Testing and conditional modification
+
+Given an initial document:
+
+[testmark]:# (test-and-conditional-modify/initial)
+```json
+{
+	"baz": "qux",
+	"foo": [
+		"a",
+		2,
+		"c"
+	]
+}
+```
+
+and a Patch OperationSequence:
+
+[testmark]:# (test-and-conditional-modify/patch)
+```json
+[
+	{ "op": "test", "path": "/baz", "value": "qux" },
+	{ "op": "test", "path": "/foo/1", "value": 2 },
+	{ "op": "add",  "path": "/bar", "value": "zar" }
+]
+```
+
+(Note that this contains both test operations as well as a subsequent add operation!
+The add operation should only apply if the test operations yield true.)
```

Comment:
this note suggests the need for a follow-up test case where it doesn't yield true, to make a test runner actually test `test` operations and not just the `add`

I reckon just copy and paste these 3 but change the second `test` value to `3` and then the last fixture shouldn't have `bar`

Reply:
I tried adding this and the result is a failing test case with `patch_test.go:63: patch did not apply: test failed`. Does testmark have a "should fail" flag of some sort?

Reply:
Gonna add a thing where if the test name ends in `-fail`, it'll be expected to fail.